src/TaskDetailForm.js

```diff
@@ -64,6 +64,7 @@ function TaskDetailForm({ task, addTask, saveTask, markComplete, categories, tag
       navigate("/");
     }
     setHasChanges(false);
+    goBack();
```

Comment:
I've tried adding a new task. After clicking "Save Task", I got an alert "Please save changes before leaving." And then the page immediately redirected to the task list page.
Probably related to this goBack() call.